Tasks/PublishPipelineArtifactV1/task.json

```diff
@@ -38,6 +38,48 @@
             "defaultValue": "",
             "required": false,
             "helpMarkDown": "Name of the artifact to publish. If not set, defaults to a unique ID scoped to the job."
+        }, 
+        {
+            "name": "artifactType",
+            "aliases": [
+                "publishLocation"
+            ],
+            "type": "pickList",
+            "label": "Artifact publish location",
+            "defaultValue": "pipeline",
+            "required": true,
```

Comment:
If a new field is required:true, this will be a breaking change, right? For yaml based pipelines?

Reply:
Please confirm that this does not break YAML based pipelines before merging.

---
In reply to: [333774443](https://github.com/microsoft/azure-pipelines-tasks/pull/11114#discussion_r333774443) [](ancestors = 333774443)

Reply:
My understanding that the YAML parsing will break customers since old YAML pipelines without this field will break.

---
In reply to: [334682981](https://github.com/microsoft/azure-pipelines-tasks/pull/11114#discussion_r334682981) [](ancestors = 334682981,333774443)

Reply:
If this change breaks existing YAML (that doesn't have these fields filled in), you will need a new major version change

Reply:
The required setting in the task repo only make the filed required in the task UI. For YAML, it will reach the agent directly. And I have tested on Dev Fabric again. Yes, YAML is still working fine.